typescript/cli/src/fork/fork.ts

```diff
@@ -50,17 +57,27 @@ export async function runForkCommand({
     forkConfig,
     readYamlOrJson,
   );
+  const chainMetadata: ChainMap<{ rpcUrls: { http: string }[] }> = {};
   for (const chainName of filteredChainsToFork) {
-    await forkChain(
+    const endpoint = await forkChain(
       context.multiProvider,
       chainName,
       port,
       kill,
       parsedForkConfig[chainName],
     );
+    chainMetadata[chainName] = { rpcUrls: [{ http: endpoint }] };
 
     port++;
   }
+
+  const mergedRegistry = new MergedRegistry({
+    registries: [registry, new PartialRegistry({ chainMetadata })],
+  });
+  const httpRegistryServer = new HttpServer(async () => mergedRegistry);
+  const httpServerPort = basePort - 10;
+  assert(httpServerPort > 0, 'HTTP server port must be greater than 0');
+  await httpRegistryServer.start(httpServerPort);
```

Comment:
Shouldn't we handle the case where the `--kill` flag is provided to the command too? I think that the server might keep running in the background if not stopped explicitly (i might be wrong)